frappe/types/filter.py

```diff
@@ -0,0 +1,280 @@
+				if (
+					not isinstance(s[0], FilterTuple)
+					and isinstance(s[0], Sequence)
+					and not isinstance(s[0][0], str)  # it's a FilterTupleSpec
+					and isinstance(s[0][0], Sequence | Mapping)
+				):
+					self.extend(
+						cast(Iterable[FilterTuple | FilterTupleSpec | FilterMappingSpec], s[0]), doctype
+					)
+				else:
+					self.extend(cast(Iterable[FilterTuple | FilterTupleSpec | FilterMappingSpec], s), doctype)
+			else:
+				self.extend(cast(Iterable[FilterTuple | FilterTupleSpec | FilterMappingSpec], s), doctype)
+		except Exception as e:
+			error_lines = str(e).split("\n")
+			indented_error = error_lines[0] + "\n" + textwrap.indent("\n".join(error_lines[1:]), "       ")
+			error_context = (
+				f"Error creating Filters:\n"
+				f"Input: {s}, doctype={doctype}\n"
+				f"Error: {indented_error}\n"
+				f"Usage: Filters( FilterTuple(...), ...                                    )\n"
+				f"       Filters( (fieldnam, value), ...                        doctype=dt )\n"
+				f"       Filters( (fieldname, operator, value), ...             doctype=dt )\n"
+				f"       Filters( (doctype, fieldname, operator, value), ...               )\n"
+				f"       Filters( {{'fieldname': value, ...}}, ...                doctype=dt )\n"
+				f"       Filters( {{'fieldname': (operator, value), ...}}, ...    doctype=dt )"
+			)
+			raise ValueError(error_context) from e
+
+		if self:  # only optimize non-empty; avoid infinit recursion
+			self.optimize()
+
+		if __debug__:
+			print(self)
+
+	@override
+	def extend(
+		self,
+		values: Iterable[FilterTuple | FilterTupleSpec | FilterMappingSpec],
+		doctype: Doct | Sentinel = UNSPECIFIED,
+	) -> None:
+		for item in values:
+			self.append(item, doctype)
+
+	@override
+	def append(
+		self, value: FilterTuple | FilterTupleSpec | FilterMappingSpec, doctype: Doct | Sentinel = UNSPECIFIED
+	) -> None:
+		if isinstance(value, FilterTuple):
+			super().append(value)
+		elif isinstance(value, Mapping):
+			if not is_not_unspecified(doctype) or doctype is None:
+				raise ValueError("When initiated with a mapping, doctype keyword argument is required")
+			self._init_from_mapping(value, doctype)
+		elif isinstance(value, Sequence):  # type: ignore[redundant-expr]
+			super().append(FilterTuple(value, doctype=doctype))
+		else:
+			raise TypeError(f"Expected FilterTruple, Mapping or Sequence, got {type(value).__name__}")
+
+	def _init_from_mapping(self, s: FilterMappingSpec, doctype: Doct) -> None:
+		for k, v in s.items():
+			if isinstance(v, _InVal):
+				self.append(FilterTuple(doctype=doctype, fieldname=k, value=v))
+			elif isinstance(v, Sequence):  # type: ignore[redundant-expr]
+				self.append(FilterTuple(doctype=doctype, fieldname=k, operator=v[0], value=v[1]))
+			else:
+				raise ValueError(f"Invalid value for key '{k}': expected value or (operator, value[s]) tuple")
+
+	def optimize(self) -> None:
+		"""Optimize the filters by grouping '=' operators into 'in' operators where possible."""
```

Comment:
This is consistent with prior implementation on the query engine and expands this other uses of filters.